flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/DefaultResourceAllocationStrategy.java

```diff
@@ -95,7 +96,7 @@ public DefaultResourceAllocationStrategy(
                 SlotManagerUtils.generateDefaultSlotResourceProfile(
                         totalResourceProfile, numSlotsPerWorker);
         this.availableResourceMatchingStrategy =
-                evenlySpreadOutSlots
+                taskManagerLoadBalanceMode == TaskManagerLoadBalanceMode.SLOTS
```

Comment:
Hi, @KarmaGYZ Thanks a lot for the review.

This part is intended to be addressed in FLINK-33388.
I have to admit that in the current PR, perhaps the balanced allocation strategy option `TASKS `for Slot latitude has been turned on too early, because there are two issues:

- We have opened the functions corresponding to the incomplete `TASKS` option. If we enable the `TASKS` option, there will be a situation where the allocation of Slot latitude is balanced, but the allocation of Slot to TM is not balanced, although this has some improvements compared to the old strategy.
- If the entire FLIP process progresses slowly, it may affect the release of the version

Based on the above questions, can one of the following ways be considered to advance the current work

- Ensure that the functionality is complete. Currently, only relevant functional codes have been added to the PR, we can disable the entry with the configuration item TASKS temporarily. It will be opened again when FLINK-33388 is pushed forward.
- Alternatively, the work content in FLINK-33388 can also be included in the current PR. This may increase the workload of the reviewer due to too many changes.

I'm willing to hear any opinions and suggestions about this.
Thank you~  :)

Reply:
I'm lean to introduce `TASKS` after the functionality is complete.

Reply:
It's fine for me. 

If so, this PR just introduces `TaskBalancedPreferredSlotSharingStrategy` related code and doesn't use them, right?

Reply:
yes, that's right.

Reply:
It's fine for me.